### DISCUSSION TITLE
How to generate different grain sizes and homogeneity of grain size in Neper

### DISCUSSION THREAD

**Discussion by Ali141272** [neperfepx/neper#655] [Q&A] [answered]
Hello Everyone,
How to generate different grain sizes model such as Fine grain (FG), Medium grain (MG0 and Coarse grain (CG)? as shown in Fig.1
The description of Fig.1 is as follow:
The size of the models is 22.5 × 45 mm. There is total 500 number of grains in the model as shown in Fig.1.Three different specimens are generated by varying the average edge length in Voronoi tessellation procedure (Fig. 1), which are 1, 1.5 and 2 mm for FG, MG and CG, respectively.

Homogeneity of Grain Size
Is it possible to generate homogeneity of grain size in Neper as you can see in Fig.2?
The sorting coefficient (So) is employed in the same model as shown in Fig.1, to represent the degree of homogeneity of the grain size distribution.

Thank you.
![Picture1](https://user-images.githubusercontent.com/102291665/223442932-a76d0b22-78e3-488d-a71f-971dca60c026.png)
![Picture2](https://user-images.githubusercontent.com/102291665/223442951-0492392c-656e-432c-8314-711192db260d.png)

**Comment by rquey** [comment, marked as answer]
Here would be the typical command for top (c):
```
neper -T -domain "square(22.5,45)" -dim 2 -morpho "diameq:lognormal(2,0.2),1-sphericity:lognormal(0.1,0.03)" -n from_morpho -o micro
neper -V micro.tess -datacellcol white -dataedgecol "20:180:20" -dataedgerad 0.07 -print img
```
![image](https://user-images.githubusercontent.com/2659376/223460778-5d3b169c-632a-4098-b958-eabc89f44f95.png)

Just vary the average cell size and sphericity values (and standard deviations) to obtain the actual figures.

**Reply by Ali141272** [reply]
Thank you very much Professor for your kind response.
Professor, is it possible to generate sorting coefficient (So) which represents degree of homogeneity of the grain size in Neper as you can see in Fig.2? The values of So are: a) So =1.05; b) So =1.09; c) So =1.13.

**Reply by rquey** [reply]
The sorting coefficient is related to the standard deviation, so, yes, using the standard deviation. (You'll have to look for the exact relationship.)

**Reply by Ali141272** [reply]
Noted Professor, Thanks once again

**Reply by Ali141272** [reply]
Professor, for generating sorting coefficient (So)/degree of homogeneity you have mentioned that “The sorting coefficient is related to the standard deviation, so, yes, using the standard deviation (You'll have to look for the exact relationship)”.
Professor now I have two type standard deviations which are;
diameq:lognormal(1,0.35) and 1-sphericity:lognormal(0.145,0.03), so which standard deviation I will use for sorting coefficient (So)/degree of homogeneity let me know.
Professor, if you have time, please give me one example using the following any sorting coefficients:
a) So =1.05; b) So =1.09

Thank you.

**Reply by rquey** [reply]
I don't use sorting coefficients, but (as I said) the concept seems related to the standard deviation. I can't help here - this is not really a Neper question.

**Reply by Ali141272** [reply]
Thank you Professor for your valuable  time.

**Reply by Mahdieh-karami** [reply]
Hi Professor,
I was wondering if it is possible to generate a hexagonal grain structure using Neper. If so, could you please advise on the appropriate commands or methodology?

**Reply by dilekguzel** [reply]
Hi @Mahdieh-karami,
I have been experimenting with ways to generate hexagonal grain structures in Neper. While manual seed placement is the most definitive way to achieve a perfect honeycomb lattice, I found that adjusting the sphericity is a useful alternative for generating highly regular structures.

Using the command below, I was able to produce a tessellation dominated by regular polygons. Although it still contains a mix of pentagons and hexagons, it is a significant step toward a hexagonal morphology:
`neper -T -n 16 -dim 2 -morpho "1-sphericity:lognormal(0.02,0.06)" -periodicity all -format geo`

<img width="833" height="797" alt="image" src="https://github.com/user-attachments/assets/6edf898f-1364-424b-a531-b2c7974065bd" />

Dilek

**Reply by rquey** [reply]
https://github.com/neperfepx/neper/discussions/1057#discussioncomment-16766427
